mobly/controllers/android_device_lib/jsonrpc_client_base.py

```diff
@@ -271,6 +260,20 @@ def _rpc(self, method, *args):
             raise ProtocolError(ProtocolError.MISMATCHED_API_ID)
         return result['result']
 
+    def _is_app_running(self):
```

Comment:
It seems odd that we can't make improvements to snippet lib because the two clients share a base class with sl4a client. If the situation demands, wouldn't it make sense to just stop sharing a base class if the clients are different enough?

We are not trying to keep the protocol identical with sl4a forever right? At one point we'll have some features that sl4a doesn't have and we shouldn't shoot down new features just to be compatible with sl4a? SL4A never had a good way to check if the connection to server is alive and well, we'd want to fix such a problem at one point, right?

But in this particular case, to preserve the inheritance, can we move `is_app_running` down to subclasses and have a different implementation? Also perhaps `is_app_running` should be named sth like `is_server_ready`?